base/server/python/pki/server/deployment/__init__.py

```diff
@@ -5376,15 +5376,135 @@ def spawn_acme(self):
+
+    def configure_est_realm(self, subsystem):
+        logger.info('Configuring EST realm')
+
+        if self.mdict['est_realm_custom']:
+            subsystem.replace_realm_config(self.mdict['est_realm_custom'])
+            return
+
+        realm_type = self.mdict['est_realm_type']
+        props = subsystem.get_realm_config(realm_type=realm_type)
+
+        if realm_type == 'in-memory':
+
+            username = self.mdict.get('est_realm_username')
+            pki.util.set_property(props, 'username', username)
+
+            password = self.mdict.get('est_realm_password')
+            pki.util.set_property(props, 'password', password)
+
+        elif realm_type == 'ds':
+
+            url = self.mdict.get('est_realm_url')
+            pki.util.set_property(props, 'url', url)
+
+            auth_type = props.get('authType')
+            auth_type = self.mdict.get('est_realm_auth_type', auth_type)
+            pki.util.set_property(props, 'authType', auth_type)
+
+            if auth_type == 'BasicAuth':
+                bind_dn = self.mdict.get('est_realm_bind_dn')
+                pki.util.set_property(props, 'bindDN', bind_dn)
+
+                bind_password = self.mdict.get('est_realm_bind_password')
+                pki.util.set_property(props, 'bindPassword', bind_password)
+
+            elif auth_type == 'SslClientAuth':
+                nickname = self.mdict.get('est_realm_nickname')
+                pki.util.set_property(props, 'nickname', nickname)
+
+            users_dn = self.mdict.get('est_realm_users_dn')
+            pki.util.set_property(props, 'usersDN', users_dn)
+
+            groups_dn = self.mdict.get('est_realm_groups_dn')
+            pki.util.set_property(props, 'groupsDN', groups_dn)
+
+        elif realm_type == 'postgresql':
+
+            url = self.mdict.get('est_realm_url')
+            pki.util.set_property(props, 'url', url)
+
+            user = self.mdict.get('est_realm_user')
+            pki.util.set_property(props, 'user', user)
+
+            password = self.mdict.get('est_realm_password')
+            pki.util.set_property(props, 'password', password)
+
+            statements = self.mdict.get('est_realm_statements')
+            pki.util.set_property(props, 'statements', statements)
+        subsystem.update_realm_config(props)
+
+    def deploy_est_webapp(self, subsystem):
+        '''
+        See also pki-server est-deploy.
+        '''
+
+        logger.info('Deploying EST webapp')
         if len(self.instance.get_subsystems()) == 1:
-            # if this is the first subsystem, start the server
-            self.instance.start(
+            # if this is the first subsystem, deploy the subsystem without waiting
+            subsystem.enable()
+
+        else:
+            # otherwise, deploy the subsystem and wait until it starts
+            subsystem.enable(
                 wait=True,
                 max_wait=self.startup_timeout,
                 timeout=self.request_timeout)
```

Comment:
The `self.instance.start()` is actually needed by `spawn_acme()` to start the instance in case ACME is the only subsystem on the server. See the test for ACME on separate instance (it's currently failing). EST probably needs a similar code too later.